src/org/etools/j1939_84/controllers/part12/Part12Step12Controller.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2021. Equipment & Tool Institute
+ */
+package org.etools.j1939_84.controllers.part12;
+
+import java.util.Arrays;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import org.etools.j1939_84.controllers.DataRepository;
+import org.etools.j1939_84.controllers.StepController;
+import org.etools.j1939_84.modules.BannerModule;
+import org.etools.j1939_84.modules.EngineSpeedModule;
+import org.etools.j1939_84.modules.VehicleInformationModule;
+
+import net.soliddesign.j1939tools.j1939.J1939DaRepository;
+import net.soliddesign.j1939tools.j1939.Lookup;
+import net.soliddesign.j1939tools.j1939.model.Spn;
+import net.soliddesign.j1939tools.j1939.packets.AcknowledgmentPacket;
+import net.soliddesign.j1939tools.j1939.packets.DM58RationalityFaultSpData;
+import net.soliddesign.j1939tools.j1939.packets.Slot;
+import net.soliddesign.j1939tools.j1939.packets.SupportedSPN;
+import net.soliddesign.j1939tools.modules.CommunicationsModule;
+import net.soliddesign.j1939tools.modules.DateTimeModule;
+
+/**
```

Comment:
The Step number is wrong here.  It should be 12.  Additionally, the description is incorrect